packages/itwin/tree-widget/src/components/trees/category-tree/CategoriesTreeComponent.tsx

```diff
@@ -174,14 +174,40 @@ function InvertButton(props: CategoriesTreeHeaderButtonProps) {
 
     const enabled: string[] = [];
     const disabled: string[] = [];
-    for (const id of ids) {
-      if (props.viewport.view.viewsCategory(id)) {
-        enabled.push(id);
+    const enabledSubCategories: string[] = [];
+    const disabledSubCategories: string[] = [];
+
+    (await props.viewport.iModel.categories.getCategoryInfo(ids)).forEach((category) => {
```

Comment:
How will this work in filtered tree? It looks like it will invert all sub categories.

Reply:
Fixed